train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -51,4 +51,16 @@ public void setJoystickPosition(int joystickPosition) {
 	public Tachograph getTg(){
 		return tg;
 	}
+
+	@Override
+	public void run() {
+		while(true){
+			followSpeed();
+			try {
+				Thread.sleep(1000);
```

Comment:
However, you should move the train at a faster rate.

Reply:
I'll lower it a bit